### PR TITLE
ARROW-9853: [RUST] Implement take kernel for dictionary arrays

### PR DESCRIPTION
**Pull request by jhorstmann**

### CODE REVIEW

**Comment by jhorstmann** [comment]
@andygrove @jorgecarleitao can I ask for your review?

**Review by jorgecarleitao — APPROVED** [review]
Thanks a lot for this PR!

LGTM: AFAI can tell the tests cover all cases.

I left a small suggestion to increase readability.

File: rust/arrow/src/compute/kernels/take.rs [hunk 1]
```diff
@@ -657,4 +694,71 @@ mod tests {
...
+
+        // dictionary values should stay the same
+        assert_eq!(&expected_values, &result_values);
+
+        let result_keys: Int16Array = result.keys().collect::<Vec<_>>().into();
+        let expected_keys = Int16Array::from(vec![
+            Some(0),
+            Some(0),
+            None,
+            Some(1),
```
[the remark this answers is not in the document]
Comment (jhorstmann) [reply rust/arrow/src/compute/kernels/take.rs:748]: Good point, I added some comments and also moved the assertion of the dictionary values closer to the assertion of the keys so hopefully their relation is now more visible.

**Review by andygrove — APPROVED** [review]
LGTM

**Review by paddyhoran — APPROVED** [review]

**Pull request closed without merging**